packages/host/app/components/operator-mode/code-submode/playground/playground-content.gts

```diff
@@ -118,17 +127,14 @@ export default class PlaygroundContent extends Component<Signature> {
             @setFormat={{this.setFormat}}
             data-test-playground-format-chooser
           />
-        {{else if (and (bool this.card) this.canWriteRealm)}}
-          <AddButton
-            class='add-field-button'
-            @variant='full-width'
-            @iconWidth='12px'
-            @iconHeight='12px'
-            {{on 'click' this.createNew}}
-            data-test-add-field-instance
-          >
-            Add Field
-          </AddButton>
+        {{else if @isFieldDef}}
+          {{#if
+            (or this.searchFieldSpec.isLoading this.createNewCard.isRunning)
+          }}
+            <LoadingIndicator @color='var(--boxel-light)' />
+          {{else if this.canGenerateFieldSpec}}
+            <div {{runFn this.createNew}} />
+          {{/if}}
```

Comment:
a little awkward. using a modifier to kickstart `createNew` if conditions have been met

Reply:
Can this be conditionally initiated in the constructor instead?

Reply:
Sure, I can make it a component